src/sinks/datadog/traces/tests.rs

```diff
@@ -116,32 +120,193 @@ fn validate_simple_span(span: dd_proto::Span) {
 
 #[tokio::test]
 async fn smoke() {
-    let mut t = simple_trace_event();
+    let mut t = simple_trace_event("a_resource".to_string());
     t.metadata_mut()
         .set_datadog_api_key(Some(Arc::from("a_key")));
 
     let events = vec![Event::Trace(t)];
     let rx = start_test(BatchStatus::Delivered, StatusCode::OK, events).await;
 
-    // We only take 1 elements as the trace & the APM transaction shall be
-    // encoded & emitted in the same payload
-    let output = rx.take(1).collect::<Vec<_>>().await.pop();
-    assert!(output.is_some());
+    // We take 2 elements as the trace & the APM transaction shall be
+    // encoded & emitted in the same payload but we also get an APM stats payload
+    let mut output = rx.take(2).collect::<Vec<_>>().await;
+
+    let trace = output.pop();
+    let stats = output.pop();
+
+    assert!(trace.is_some());
+    assert!(stats.is_some());
```

Comment:
It's guaranteed that the trace will `pop` first everytime?

Reply:
I thinks the order is preserved down to the underlying tower service. I'll double check.